src/main/java/ro/utcn/sd/mdantonio/StackUnderflow/service/PostManagementService.java

```diff
@@ -0,0 +1,141 @@
+package ro.utcn.sd.mdantonio.StackUnderflow.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import ro.utcn.sd.mdantonio.StackUnderflow.entities.*;
+import ro.utcn.sd.mdantonio.StackUnderflow.exception.InvalidActionException;
+import ro.utcn.sd.mdantonio.StackUnderflow.exception.InvalidPermissionException;
+import ro.utcn.sd.mdantonio.StackUnderflow.exception.ObjectNotFoundExpection;
+import ro.utcn.sd.mdantonio.StackUnderflow.repository.API.PostRepository;
+import ro.utcn.sd.mdantonio.StackUnderflow.repository.API.RepositoryFactory;
+import ro.utcn.sd.mdantonio.StackUnderflow.repository.API.VoteRepository;
+
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static ro.utcn.sd.mdantonio.StackUnderflow.entities.StackUnderflowConstants.*;
+
+@Service
+@RequiredArgsConstructor
+public class PostManagementService implements ManagementService {
+    private final RepositoryFactory repositoryFactory;
+    @Transactional
+    public List<Post> listQuestions(){
+        List<Post> questions =  repositoryFactory.createPostRepository().findAll().stream().
+                filter(x->x.getPosttypeid()==QUESTIONID).collect(Collectors.toList());
+        questions.forEach(x->x.setScore(calculatePostScore(x)));
+        return questions.stream().sorted(Comparator.comparing(Post::getCreationDate,
```

Comment:
Doing two sorts is redundant - only the last sort is actually applied. What were you trying to do exactly?

Reply:
So you mean if two values have identical dates and different scores the order is the same as if I sorted them only by score?

Reply:
Yes, the second sort will completely overturn the first one (except for cases where the scores are equal; in that case if the second sort is stable, you will retain some results from the first one). 

I would strongly suggest that you just do one sort, but combine both comparators into a single one. You can do this with the [Comparator#thenComparing](https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html#thenComparing-java.util.Comparator-) methods. Basically you can specify another comparator to be used in cases where the first one returns 0 (i.e. when the values are equal).